Properly delete fake cache before provisioning

mastacheata opened the pull request:
We seem to have forgotten changing the delete query to also use the provider_key.
Right now it's looking for self.provider and won't find anything to delete, which in turn will cause an error with the UniqueConstraint for provider/provider_offset

fixes #16

mastacheata commented:
Hmm, seems like I should rebase my branch again.

costela commented on django_scrubber/scrubbers.py:
```diff
@@ -130,9 +130,9 @@ def _initialize_data(self):
             faker_instance.add_provider(provider)
 
-        logger.info('Initializing fake scrub data for provider %s' % self.provider)
+        logger.info('Initializing fake scrub data for provider %s' % self.provider_key)
```
The `provider_key` here is not very helpful, since it has a hash of arguments. Mabye just use the `self.provider` and the args' textual representation?

mastacheata replied:
Not sure if I did this right. Just did this on the fly without having run the testsuite on my machine yet. If it fails I'll try more thoroughly tomorrow.

costela commented:
Thanks @mastacheata, LGTM.
But could you please rebase your changes onto master?

mastacheata commented:
> 
> 
> Thanks @mastacheata, LGTM.
> But could you please rebase your changes onto master?

Just did that. It just seems our branches have diverged too much?
I'll try something stupid and create a new clean branch from your matser and will only apply the 4 commits I did for this issue.

mastacheata commented:
It looks like Github might have a problem picking up with what I did just now.

created a new branch from master
cherry-picked the 4 commits in this new PR
squashed them in pairs of 2 so my mistake-fix combos would appear as only one commit each

The result should be my branch being ahead of your master by 2 commits.

OK, I rebased my own master on your master and pushed everything, now it seems to have picked up my changes.
